CUDADataFormats/BeamSpot/interface/BeamSpotCUDA.h

```diff
@@ -0,0 +1,44 @@
+#ifndef CUDADataFormats_BeamSpot_interface_BeamSpotCUDA_h
+#define CUDADataFormats_BeamSpot_interface_BeamSpotCUDA_h
+
+#include <cuda_runtime.h>
+
+#include "HeterogeneousCore/CUDAUtilities/interface/device_unique_ptr.h"
+
+class BeamSpotCUDA {
+public:
+  // align to the CUDA L1 cache line size
+  struct alignas(128) Data {
```

Comment:
In the previous incarnation of this PR (#29154) there was a [comment](https://github.com/cms-sw/cmssw/pull/29154#issuecomment-596692163) from @slava77 of which couldn't find any conclusion in the thread: 
"I think that it would be useful to have this struct in regular DataFormats/BeamSpot as BeamSpotPOD or BeamSpotCore or something like that.
This way we could even make the "legacy" producer to write this, if it becomes useful."

Reply:
on a related subject, is it possible to do edm::Event::get of this BeamSpotCUDA from a regular ED producer on CPU?

Reply:
> I think that it would be useful to have this struct in regular DataFormats/BeamSpot as BeamSpotPOD or BeamSpotCore or something like that. This way we could even make the "legacy" producer to write this, if it becomes useful.

Sorry, I'm afraid I missed that.

In general, sure, it could make sense.

Specifically for the beam spot maybe not, since we are finally going in the direction of putting the beam spot in the conditions also at HLT, so we could eventually get ride of the event product.

Of course, we can do it also for the beam spot, and eventually remove it if/when it has no use any more.

Reply:
> on a related subject, is it possible to do edm::Event::get of this BeamSpotCUDA from a regular ED producer on CPU?

I think so, as long as it uses the proper context; see the [Producer with CUDA input](https://github.com/cms-sw/cmssw/blob/master/HeterogeneousCore/CUDACore/README.md#producer-with-cuda-input) and [Analyzer with CUDA input](https://github.com/cms-sw/cmssw/blob/master/HeterogeneousCore/CUDACore/README.md#analyzer-with-cuda-input) examples.

Of course the module will need to use the proper semantic and request the copy of the data form the GPU to the CPU.
A better pattern would be to write a dedicated "FromCUDA" `EDProducer` that takes care of reading the GPU object, copy it to the CPU memory, and make it available in the Event.